js.go

```diff
@@ -1096,6 +1091,12 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync
 		return nil, err
 	}
 
+	// With flow control enabled async subscriptions we will disable msgs
+	// limits, and set a larger pending bytes limit by default.
+	if !isPullMode && cb != nil && hasFC {
+		sub.SetPendingLimits(DefaultSubPendingMsgsLimit*4, DefaultSubPendingBytesLimit*4)
```

Comment:
Would leave Bytes alone, but upping messages ok by me.

Reply:
The current limit is:
```
DefaultSubPendingMsgsLimit*4  ->   262,144 pending messages
```

Maybe we could pick one of these instead?

```
DefaultSubPendingMsgsLimit*8  ->   524,288 pending messages
DefaultSubPendingMsgsLimit*16 -> 1,048,576 pending messages
```

Reply:
Well its the most amount of properly FC'd messages within 1MB that we could receive properly. *16 is good I think..

Reply:
Updated message pending limit. 👍🏽 
```diff
-sub.SetPendingLimits(DefaultSubPendingMsgsLimit*4, DefaultSubPendingBytesLimit*4)
+sub.SetPendingLimits(DefaultSubPendingMsgsLimit*16, DefaultSubPendingBytesLimit*4)
```